src/verify_cert.rs

```diff
@@ -341,11 +341,23 @@ fn check_basic_constraints(
     }
 }
 
+/// An OID value indicating the Extended Key Usage (EKU) of the certificate.
 #[derive(Clone, Copy, PartialEq, Eq)]
-pub(crate) struct KeyPurposeId {
+pub struct KeyPurposeId {
     oid_value: untrusted::Input<'static>,
 }
 
+impl KeyPurposeId {
+    /// Construct a new [`KeyPurposeId`].
+    ///
+    /// `oid` is the OBJECT IDENTIFIER in bytes.
+    pub fn new(oid: &'static [u8]) -> Result<Self, Error> {
```

Comment:
I think this could be written to return `Self` instead of a `Result`. The `untrusted::Input::from` is infallible. Was there a reason you were preferring `Result` but always returning `Ok`?

Reply:
No, no good reason 😅 I'll change it.